src/main/java/com/google/api/gax/core/Page.java

```diff
@@ -37,50 +39,26 @@
  * object also provides methods to retrieve additional pages using the page token, and to get the
  * API request and response objects.
  */
-public interface Page<RequestT, ResponseT, ResourceT> extends Iterable<ResourceT> {
+public interface Page<ResourceT> extends Iterable<ResourceT> {
```

Comment:
We would still have `iterateAll()` - this would only be if a user has chosen to access only the values in that page.

Discussing further with @garrettjonesgoogle, we think that sticking with `getValues()` is likely best, as we can more easily support resource name types when appropriate.

So the `Page` implementation we will end up with will live in GAX, and will provide an `iterateAll()` method that fetches all values by making additional API calls in the background, and a `getValues()` method that provides only the values in that page.